### ISSUE TITLE
How to query  map type column in cassandra.

### ISSUE THREAD

**Issue by kannuriajay** [Impetus/kundera#659]
HI  I have table with column ----
CREATE TABLE experiment_by_lab (
labid timeuuid,
procedure map<timeuuid, text>,
experiment_id timeuuid,
title text,
type text,
primary key(labid, experiment_id, title, type)
);

I need to write two method fetch records the procedure by type 
procedure by name.

so I wrote first method 

```
public Experiment getProcedureId(UUID labId,Map <UUID,String> procedure) {
    String queryStr = "select * from experiment_by_lab  where labid=? and procedure ?";
    SialDatastaxQuery<Experiment> query = entityManager.createDatastaxQuery(
            queryStr, Experiment.class);
    query.setParameter(0, labId).
    setParameter(1, procedure);
    Experiment experiment = (Experiment) query.getSingleResult();
return experiment;
```

}

How to write query to get only for procedure UUID . ?
"select \* from experiment_by_lab  where labid=? and procedure ?"
procedure map type?

Thanks, 
AjayKumar

**Comment by chhavigangwal** [comment]
@kannuriajay 

Kundera currently supports (and is most stable with) Cassandra 2.0.4.
In order to query over a field in Cassandra , you need to create an index on it. However indexing is not supported over collections in Cassandra till version 2.1 (https://issues.apache.org/jira/browse/CASSANDRA-4511)

Hence you cannot query over map type columns in Cassandra as of now.

Chhavi

**Comment by kannuriajay** [comment]
Hi,
I have upgraded DB to cassandra 2.1.1.

how to write a method: get experiment by procedure id pointing to procedure map.

In code How can I separate the Key from map. and display only key.

procedure map<timeuuid, text>,

Regards.
AjayKumar.

**Comment by chhavigangwal** [comment]
@kannuriajay 

For now Kundera only supports Cassandra 2.0.4 so you cannot query or create indexes over collections. However you can query the map using Kundera by using its native query support for now e.g. 

```
 Query q = em.createNativeQuery("select * from experiment_by_lab where procedure CONTAINS 'Procedure 1'", Experiment.class);
```

You can find more help at : 
http://www.datastax.com/dev/blog/cql-in-2-1

Hope that Helps!
Chhavi

**Comment by kannuriajay** [comment]
Hi,

We are using Datastax driver not Kundera. and cassandra 2.1.1

AjayKumar

**Comment by chhavigangwal** [comment]
@kannuriajay 

I did not get you , are you not using Kundera to perform operations on data stored in Cassandra ?

Because , if you are trying to query data stored in Cassandra 2.1.1 using Kundera, you need to create index over the map column manually and then query it using nativeQuery support as I shared in example above.

Chhavi

**Comment by chhavigangwal** [comment]
Fixed with Kundera -2.16 release.
